### ISSUE TITLE
llvm global-to-wide tests not tested nightly

### ISSUE THREAD

**Issue by mppf** [chapel-lang/chapel#11192]
`compiler/llvm/llvm-global-to-wide/test/*.ll` includes several tests of our LLVM pass global-to-wide. These should be run nightly.

Additionally it would be nice to have a strategy for testing other Chapel optimizations implemented as LLVM passes, where these optimizations can be provided LLVM IR input rather than Chapel code.

**Comment by bradcray** [comment]
If the LLVM back-end were the default, would testing these nightly simply be a matter of adding compopts flags to tests and/or firing off configurations in the playground or the like?

**Comment by mppf** [comment]
No, because these are testing a specific LLVM pass and the input is not Chapel code but rather LLVM IR. We do test that LLVM pass nightly in an end-to-end manner with Chapel code, but it can be useful to also test that pass in isolation (e.g. easier to check more cases, more stable vs changes in other parts of the Chapel language/compiler, ...).

**Comment by mppf** [comment]
@dmk42 - we were discussion the approach for this. IIRC it's:

 * add start_test support for something.test.ll files and for .filecheck files (still using .compopts to specify what to run, presumably)
 *  generate a .so with any Chapel-LLVM optimizations that can be loaded in opt

The .so file is interesting. Should there be a separate .so for each optimization or group of optimization? Should some testing system environment variable activate building the .so or should we try to build it in end-user Chapel compiler `make` calls?

**Comment by dmk42** [comment]
For debugging purposes, my guess is that the most convenient would be to have a separate .so file for each optimization or group of optimizations, but I don't have a strong preference.

My intuition is that the .so files should be generated whenever the global-to-wide tests are run (not depending on another environment variable), but I won't be disappointed if you decide the other way.
